WordPress/gutenberg
# Move theme location settings to navigation editor sidebar

## Description

Closes https://github.com/WordPress/gutenberg/issues/28864

Branched from ( https://github.com/WordPress/gutenberg/pull/29012 )
The aim of this PR is to change the placement of Manage Locations to imitate the mechanics from `/wp-admin/nav-menus.php?action=locations` (see the screenshot). 

Manage Locations button has been moved from Header to sidebar. Additionally, checkboxes have been added. Checkboxes are the indicators if a selected menu is placed in a menu location. If a checkbox relating to menu location is selected it means a selected menu is bound to this location. By design - one menu can be in multiple locations. 


## How has this been tested?



Tested manually 

In order to test:
1. Navigate to Guttenberg > Navigation (beta)
2. Click on the editor, so the navbar will appear
3. On the navbar select/unselect the checkbox with a name of menu location (primary, footer)
4. Click on the `manage locations` button
5. Modal with dropdowns will appear
6. Values in dropdowns should be equal to the name of the selected menu only if a corresponding checkbox is selected
7. If a checkbox is selected, and a user will select any other menu name in a corresponding dropdown then the checkbox should unselect
8. All states (checkboxes, dropdowns) should remain after a page refresh

## Screenshots 
![image](https://user-images.githubusercontent.com/8398557/110499660-bf9e4f00-80f8-11eb-8b15-d372c39d32f8.png)

## Types of changes




New feature 

## Checklist:
